src/azul/indexer/document.py

```diff
@@ -403,10 +403,9 @@ def translate_fields(cls,
         else:
             if isinstance(field_types, list):
                 # FIXME: Assert that a non-list field_type implies a non-list
-                #        doc (only possible for for contributions).
+                #        doc (only possible for contributions).
                 #        https://github.com/DataBiosphere/azul/issues/2689
-                if forward:
-                    assert doc is None or isinstance(doc, list)
+                assert doc in (None, NullableString.null_string) or isinstance(doc, list)
```

Comment:
To tighten the assert, it should take the direction of the translation into account.